opm/simulators/flow/MainDispatchDynamic.cpp

```diff
@@ -131,6 +133,11 @@ int Opm::Main::dispatchDynamic_()
         return this->runThermal(phases);
     }
 
+    // Blackoil case with temperature
+    else if (phases.size() == 4 && temp) {
```

Comment:
I would assume that phases.size() would be 3 not 4? Or would the energy bit be set in the Runspec Phases object? If so that is a bit counterintuitive.